aiplatform/snippets/embeddings.go

```diff
@@ -28,40 +27,34 @@ import (
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
-func embedTexts(
-	project, location string, texts []string) ([][]float32, error) {
+func embedTexts(project, location, model string) ([][]float32, error) {
 	ctx := context.Background()
 
 	apiEndpoint := fmt.Sprintf("%s-aiplatform.googleapis.com:443", location)
-	model := "text-embedding-004"
-	task := "QUESTION_ANSWERING"
-	customOutputDimensionality := 5
+	dimensionality := 5
+	texts := []string{"banana muffins? ", "banana bread? banana muffins?"}
 
 	client, err := aiplatform.NewPredictionClient(ctx, option.WithEndpoint(apiEndpoint))
 	if err != nil {
 		return nil, err
 	}
 	defer client.Close()
 
-	match := regexp.MustCompile(`^(\w+-\w+)`).FindStringSubmatch(apiEndpoint)
-	if match != nil {
-		location = match[1]
-	}
 	endpoint := fmt.Sprintf("projects/%s/locations/%s/publishers/google/models/%s", project, location, model)
 	instances := make([]*structpb.Value, len(texts))
 	for i, text := range texts {
 		instances[i] = structpb.NewStructValue(&structpb.Struct{
 			Fields: map[string]*structpb.Value{
 				"content":   structpb.NewStringValue(text),
-				"task_type": structpb.NewStringValue(task),
+				"task_type": structpb.NewStringValue("QUESTION_ANSWERING"),
```

Comment:
nit: I would keep the task_type as a string literal initialized outside of this object. Up to you!

Reply:
Thanks, for your comments! 
In this case I don't have strong preference, but received comments in previous PRs, to not create variables, which used only once. Therefore I am trying to follow consistent pattern

Reply:
Okay, fair enough. I'll chat with the other Golang reviewers about what our preferred style should be.